File: solution/226.invert-binary-tree/invert-binary-tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
void reverse(struct TreeNode*root){
    if(!root){
        return;
    }
    reverse(root->left);
    reverse(root->right);
    struct TreeNode*temp = root->left;
    root->left = root->right;
    root->right = temp;
}
struct TreeNode* invertTree(struct TreeNode* root) {
    reverse(root);
    return root;
}

File: solution/226.invert-binary-tree/invert-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
function reverse(root) {
        if (!root) {
          return;
        }
        reverse(root.left);
        reverse(root.right);
        let temp = root.left;
        root.left = root.right;
        root.right = temp;
      }
      var invertTree = function(root) {
        reverse(root);
        return root;
      };